### PR TITLE
Atmega adc support

### PR DESCRIPTION
**Pull request by mali**
This PR improve and obsolete PR #6148 by adding adc support to atmega based boards.
For now it covers atmega328p and atmega2560, atmega1281 should be trivial to add.
atmega2560 is only build tested, so it needs real testing by someone having the needed hardware.
* Prescaler is fixed to /128 (125kHz)
* Vref is fixed to 5V
Also, it needs PR #6581 to build properly.

### CODE REVIEW

**Comment by PeterKietzmann** [comment]
@mali please close the other PR in case it's obsolete and point to this PR as a last comment.

**Comment by mali** [comment]
How to trigger a new jenkins built without pushing commit ?

**Comment by kYc0o** [comment]
Well the errors are on the mq3 driver, which was merged afaik... maybe a rebase will be enough to include the changes?

To retrigger a build you need to login with the github account into Jenkins, then you can start a new building.

**Comment by mali** [comment]
@kYc0o Sorry, haven't found how to force rebuild on Jenkins. I can login, but maybe missing rights ...

**Comment by aabadie** [comment]
@mali I just rescheduled a build.

**Comment by aabadie** [comment]
As @kYc0o said, you should be able to restart a build from Jenkins yourself : Click on Details to go to Jenkins, then click "Back to project" and then click "Build now".

**Comment by mali** [comment]
@aabadie thank's.
I can confirm I don't have this "Build now" option.

**Comment by aabadie** [comment]
@mali, you need to be logged in with your GitHub account (just to be sure ;) ). Otherwise I think this can be because your are not part of the maintainers.

**Comment by mali** [comment]
I don't know how to solve in an elegant way last build failures.
I've added arduino adc API implementation into /sys/arduino but now, boards that have arduino feature enabled but no adc support failed to build. 
I've tried adding FEATURES_REQUIRED = perip_adc in sys/arduino Makefile, also tried to mark arduino adc API with #ifdef FEATURE_PERIPH_ADC but without success. 
remove arduino feature for these boards ? I don't like the idea ...
any hints ?

**Comment by kYc0o** [comment]
I'm testing this PR with arduino-uno without success... Did you try with tests/periph_adc ?

**Comment by mali** [comment]
no, I've done my own tests, will do with tests/periph_adc ...

**Comment by mali** [comment]
I suspect a problem when using multiple ADCs simultaneously, I had only done unit tests ...

File: boards/arduino-atmega-common/include/periph_conf.h [hunk 1]
```diff
@@ -142,6 +142,41 @@ extern "C" {
 #endif
 /** @} */
 
+/**
+ * @brief ADC configuration
+ *
+ * @{
+ */
+#ifdef CPU_ATMEGA328P
+#define ADC_NUMOF       (6)
```
Comment (cgundogan) [inline boards/arduino-atmega-common/include/periph_conf.h]: according to the datasheet, there are 8 ADCs (ADC0-7)

File: cpu/atmega_common/periph/adc.c [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        DIDR0 |= (1 << line);
+#ifdef DIDR2
+    else
+        DIDR2 |= (1 << (line-8));
+#endif
+    /* Enable ADC */
+    ADCSRA |= (1 << ADEN);
+
+    /* Enable ADC with Prescaling = /128 = 125KHz */
+    ADCSRA=(1 << ADEN)|(1 << ADPS2)|(1 << ADPS1)|(1 << ADPS0);
```
Comment (cgundogan) [inline cpu/atmega_common/periph/adc.c]: no need to include (1 << ADEN) here (as you did above)

File: cpu/atmega_common/periph/adc.c [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    /* Ref Voltage is Vcc(5V) */
+    ADMUX = (1 << REFS0);
```
Comment (cgundogan) [inline cpu/atmega_common/periph/adc.c]: for completeness: unset REFS1

File: boards/arduino-atmega-common/include/periph_conf.h (hunk 1 above)
Comment (smlng) [inline boards/arduino-atmega-common/include/periph_conf.h]: au contraire, Arduino Duemilanove has only 6 analog in pins accessible, though there may be 8 internally. See [here](https://www.arduino.cc/en/Main/ArduinoBoardDuemilanove), picture and reference table.

**Comment by aabadie** [comment]
@mali, any progress on this one ? It would be great if you could address the comments reported ;)

**Comment by mali** [comment]
@aabadie, yes I plan to rework on it soon. I need to update my debian to new stable and so switch to gcc6, and with this I must be able to compile Riot again ;-)

**Comment by dnahm** [comment]
I already reworked it. The ADC driver covers now atmega328p, atmega2560 and atmega1281. The driver was tested on each listed Microcontroller. The PR will be come in the next days.

Atmega328p with the PDIP package has 6 ADCs (Arduino UNO). But only 5 ADCs can be used, because the pin of ADC5 emulate a software triggered interrupt in RIOT.

**Comment by dnahm** [comment]
New PR: cpu/atmega_common: add adc driver #7227

**Comment by mali** [comment]
Closed in favor of PR #7227

**Pull request closed without merging**